x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -468,6 +469,31 @@ export class Embeddable
         this.input.onTableRowClick(event.data as unknown as LensTableRowContextMenuEvent['data']);
       }
     }
+
+    // TODO - inject this from datatable_visualization/visualization
+    const onEditAction = (state, event) => {
+      switch (event.data.action) {
+        case 'sort':
+          return {
+            ...state,
+            sorting: {
+              columnId: event.data.columnId,
+              direction: event.data.direction,
+            },
+          };
+        };
+    };
```

Comment:
@flash1293 Declaring the handler here is just to prove out the concept and needs to be changed.

I'm not sure of the right approach. This `Embeddable` class is more generic than just datatables. So, do we export the real [`onEditAction` handler](https://github.com/andrewctate/kibana/blob/30d6a122a16693ef8c0d05e86d364e0a42d27c16/x-pack/plugins/lens/public/datatable_visualization/visualization.tsx#L402-L444) and provide it with the rest of the `Embeddable`'s dependencies? If so, I don't see how we can know that we're working with a datatable in the `EmbeddableFactory`.

It feels like I'm on the wrong track. 🤷‍♂️

Reply:
The editorframe service has the concept of the "visualizationMap" - this is an object with all visualization definitions. We could expose it there and it pass down to the embeddable to use the right edit action handler in a generic way.